apps/fft/fft-stdexec.cpp

```diff
@@ -0,0 +1,221 @@
+
+//
+// fft algorithm
+//
+std::vector<data_t> fft(data_t *x, scheduler auto sch, const int N, const int max_threads, bool debug = false)
+{
+    std::vector<data_t> x_rev(N);
+    std::vector<uint32_t> ind(N);
+
+    data_t *x_r = x_rev.data();
+    uint32_t *id = ind.data();
+
+    int shift = 32 - ilog2(N);
+
+    ex::sender auto twiddle = ex::transfer_just(sch, x_r, x, id)
+        | ex::bulk(N, [=](int k, auto x_r, auto x, auto id){
+            id[k] = reverse_bits32(k) >> shift;
+            x_r[k] = x[id[k]];
+        })
+        | ex::then([](auto &&...){});
+
+    ex::sync_wait(twiddle);
+
+    // niterations
+    int niters = ilog2(N);
+    // local merge partition size
+    int lN = 2;
+
+    // set cout precision
+    std::cout << std::fixed << std::setprecision(1);
+    std::cout << "FFT progress: ";
+
+    // transfer_just sender
+    ex::sender auto tx = ex::transfer_just(sch, x_r);
+
+    for (int k = 0; k < niters; k++, lN*=2)
+    {
+        std::cout << (100.0 * k)/niters << "%.." << std::flush;
+
+        static Timer dtimer;
+
+        // number of partitions
+        int nparts = N/lN;
+        int tpp = lN/2;
+
+        if (debug)
+        {
+            dtimer.start();
+            std::cout << "lN = " << lN << ", npartitions = " << nparts << ", partition size = " << tpp << std::endl;
+        }
+
+        ex::sender auto merge = tx | ex::bulk(N/2, [=](auto k, auto y)
+        {
+            // compute indices
+            int  e   = (k/tpp)*lN + (k % tpp);
+            auto o   = e + tpp;
+            auto i   = (k % tpp);
+            auto tmp = y[e] + y[o] * WNk(N, i * nparts);
+            y[o]     = y[e] - y[o] * WNk(N, i * nparts);
+            y[e]     = tmp;
+        });
+
+        ex::sync_wait(std::move(merge));
+
+        if (debug)
+            std::cout << "This iter time: " << dtimer.stop() << " ms" << std::endl;
+    }
+
+    std::cout << "100%" << std::endl;
+
+    return x_rev;
+}
+
+//
+// simulation
+//
+int main(int argc, char* argv[])
+{
+    // parse params
+    fft_params_t args = argparse::parse<fft_params_t>(argc, argv);
+
+    // see if help wanted
+    if (args.help)
+    {
+        args.print();  // prints all variables
+        return 0;
+    }
+
+    // simulation variables
+    int N = args.N;
+    sig_type_t sig_type = sig_type_t::box;
+    int max_threads = args.max_threads;
+    //int freq = args.freq;
+    bool print_sig = args.print_sig;
+    bool print_time = args.print_time;
+    bool validate = args.validate;
+    std::string sched = args.sch;
+
+    // x[n] signal
+    sig_t x_n(N, sig_type);
+
+    if (!isPowOf2(N))
+    {
+        N = ceilPowOf2(N);
+        std::cout << "INFO: N is not a power of 2. Padding zeros => N = " << N << std::endl;
+
+        x_n.resize(N);
+    }
+
+    if (print_sig)
+    {
+        std::cout << std::endl << "x[n] = ";
+        x_n.printSignal();
+        std::cout << std::endl;
+    }
+
+    // y[n] = fft(x[n]);
+    std::vector<data_t> y;
```

Comment:
do we want to resize it? it seems it is default un-initialized

Reply:
fixed.